### PR TITLE
Dbname to process

### PR DESCRIPTION
**Pull request by ManjotS**

### CODE REVIEW

File: api/config/index.js [hunk 1]
```diff
@@ -7,29 +40,36 @@ const metadata = {
...
+// if npm run local, then just define what cli gives, otherwise use process.json
+if (!!process.env.DB_NODES) {
+  const dbs = JSON.parse(process.env.DB_NODES);
+  dbs.forEach(db => {
+    const thisObj = {
+      ...metadata,
+      host: db.host,
+      port: db.port || 3306,
+    };
+    if (!!db.isPrimary && db.isPrimary === true) {
```
Comment (tsingh777) [inline api/config/index.js:58]: not needed in javascript, if `isPrimary` is set to `true`,  `!!db.isPrimary` will be truthy.

File: app.js [hunk 2]
```diff
@@ -11,10 +11,7 @@ const port = process.env.NODE_ENV === 'development' ? '3001' : '3000';
 
 // database
 app.locals.pool = createPoolCluster();
-app.locals.pool.add('local', config.mysql0);
-app.locals.pool.add('db1', config.mysql1);
-app.locals.pool.add('db2', config.mysql2);
-app.locals.pool.add('db3', config.mysql3);
+config.forEach((db, dsn) => app.locals.pool.add(db, dsn));
```
Comment (tsingh777) [inline app.js:14]: should use more descriptive variable names. 

dsn here would be the index. I don't think that's what you meant to pass to `pool.add`. Think you want to do 
```
config.forEach((dbConfig, index) => app.locals.pool.add(`db${index}`, dbConfig));

```

File: api/config/index.js [hunk 3]
```diff
@@ -1,3 +1,36 @@
+/**
+ * Expects process-dev.json or process.json (or run local)
+ *
+ * @example
```
Comment (tsingh777) [inline api/config/index.js:4]: 👍

File: app.js (hunk 2 above)
Comment (ManjotS) [reply app.js:14]: it doesn't seem to work that way, it seems to be index in db. I tested this.

Comment (ManjotS) [reply app.js:14]: dsn = database source name or basically the db config itself

Comment (ManjotS) [reply app.js:14]: should be used as pool.add("name of connection", "config obj")

Comment (ManjotS) [reply app.js:14]: looked at docs, dont understand why this works as is.

Comment (ManjotS) [reply app.js:14]: ok tried it with 

config.forEach((dsn, i) => app.locals.pool.add(i, dsn));

and it doesn't work.

Comment (tsingh777) [reply app.js:14]: configArry is an array populated with an object. 

```
configArry = [{
     user: process.env.DB_USER || 'root',
     password: process.env.DB_PASSWORD || 'root',
     database: process.env.NODE_ENV === 'development' ? 'khajana_dev_khajana' : 'khajana_khajana',	
     dateStrings: true,	 
     acquireTimeout: 5000,
     connectionLimit: process.env.DB_POOL_SIZE,
     host: db.host,
     port: db.port || 3306,
}];
```

looping over it with foreach takes a [callback](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#Description) with params (value, index). 

Are your sure this is working?

Comment (tsingh777) [reply app.js:14]: do 

```
config.forEach((dsn, i) => app.locals.pool.add(`db${i}`, dsn));
```

**Pull request merged**